src/main/java/Task.java

```diff
@@ -0,0 +1,103 @@
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Task object for CS2103T iP. Stores relevant data regarding a task, including type, task,
+ * date and time (if required) and whether it has been completed.
+ */
+public class Task {
+    private final String task;
+    private boolean done = false;
+    private final int type;
+    private final LocalDate date;
+    private final LocalTime time;
+
+    /**
+     * Creates a Task object.
+     * @param task Description of task.
+     * @param type Type of task: 0 is todo, 1 is deadline, 2 is event
+     */
+
+    public Task(String task, int type) {
+        if (type == 0) {
+            this.task = task;
+            this.date = null;
+            this.time = null;
+        } else {
+            String[] info = task.split("/");
+            if (info.length == 1 || info[0].equals(" ")) {
+                throw new IllegalArgumentException();
+            } else if (type == 1) {
+                this.date = LocalDate.parse(info[1].substring(3));
+                this.time = null;
+            } else {
+                this.date = null;
+                this.time = LocalTime.parse(info[1].substring(3));
+            }
+            this.task = info[0];
+        }
+        this.type = type;
+    }
+
+    /**
+     * When called on a task, marks the task as done.
+     */
+    public void markDone() {
+        this.done = true;
+    }
+
+    /**
+     * Checks for whether the task is done. Used as part of the toString() method.
+     * @return "[ ]" for not done, "[X]" for done.
+     */
+    public String checkDone() {
```

Comment:
Perhaps you should rename the method as the name does not reflect its behaviour?